src/cargo/ops/cargo_install.rs

```diff
@@ -179,15 +180,32 @@ impl<'gctx> InstallablePackage<'gctx> {
             }
         };
 
-        let (ws, rustc, target) =
-            make_ws_rustc_target(gctx, &original_opts, &source_id, pkg.clone())?;
-        // If we're installing in --locked mode and there's no `Cargo.lock` published
-        // ie. the bin was published before https://github.com/rust-lang/cargo/pull/7026
-        if gctx.locked() && !ws.root().join("Cargo.lock").exists() {
-            gctx.shell().warn(format!(
-                "no Cargo.lock file published in {}",
-                pkg.to_string()
-            ))?;
+        let (ws, rustc, target) = make_ws_rustc_target(
+            gctx,
+            &original_opts,
+            &source_id,
+            pkg.clone(),
+            lockfile_path.clone(),
+        )?;
+
+        if gctx.locked() {
+            // When --lockfile-path is set, check that passed lock file exists
+            // (unlike the usual flag behavior, lockfile won't be created as we imply --locked)
+            if let Some(requested_lockfile_path) = ws.requested_lockfile_path() {
+                if (!requested_lockfile_path.exists()) {
```

Comment:
Perhaps better.



```suggestion
                if !requested_lockfile_path.is_file() {
```

After this change I believe we're ready to go :)